### PR TITLE
Chair enter types

### LINKED ISSUE #2018: Change seatspawn to allow more than just Seat_Airboat type
The roller coaster sitting animation is not necessarily desirable for every contraption.

### PR DESCRIPTION
**Pull request by Hazeofdream**
Closes #2018 

review because i didn't test these functions and i was kinda winging it tbh, My big worry is getting the wrong seat types as i couldn't find the real types for them on the wiki.

### CODE REVIEW

File: lua/entities/gmod_wire_expression2/core/custom/prop.lua [hunk 1]
```diff
@@ -26,17 +26,19 @@ function PropCore.WithinPropcoreLimits()
...
+	
+	if seat == "" then seat == "Seat_Airboat"
 
 	if not PropCore.WithinPropcoreLimits() then
 		ret = false
 	elseif not (util.IsValidProp( model ) and WireLib.CanModel(ply, model)) then
 		ret = false
 	elseif isVehicle then
-		ret = gamemode.Call( "PlayerSpawnVehicle", ply, model, "Seat_Airboat", list.Get( "Vehicles" ).Seat_Airboat ) ~= false
+		ret = gamemode.Call( "PlayerSpawnVehicle", ply, model, seat, list.Get( "Vehicles" ).seat ) ~= false
```
Comment (Vurv78) [inline lua/entities/gmod_wire_expression2/core/custom/prop.lua:41]: This syntax is incorrect. You're basically doing `list.Get("Vehicles")["seat"]`, while you're supposed to do `list.Get("Vehicles")[seat]` to use the `seat` variable

File: lua/entities/gmod_wire_expression2/core/custom/cl_prop.lua [hunk 2]
```diff
@@ -10,7 +10,9 @@ E2Helper.Descriptions["propSpawn(ean)"] = "Rotation, Frozen Spawns a prop with t
 E2Helper.Descriptions["propSpawn(svan)"] = "Model path, Position, Rotation, Frozen Spawns a prop with the model denoted by the string file path, at the position denoted by the vector, and rotated to the angle given. If frozen is 0, then it will spawn unfrozen."
 E2Helper.Descriptions["propSpawn(evan)"] = "Position, Rotation, Frozen Spawns a prop with the model of the template entity, at the position denoted by the vector, and rotated to the angle given. If frozen is 0, then it will spawn unfrozen."
 E2Helper.Descriptions["seatSpawn(sn)"] = "Model path, Frozen Spawns a prop with the model denoted by the string filepath. If frozen is 0, then it will spawn unfrozen."
+E2Helper.Descriptions["seatSpawn(sns)"] = "Last string refers to seat type, Valid types are Seat_Airboat, Seat_Pod and Seat_Jeep"
```
Comment (Vurv78) [inline lua/entities/gmod_wire_expression2/core/custom/cl_prop.lua:13]: Pretty sure there are more valid types than those: https://github.com/Facepunch/garrysmod/blob/master/garrysmod/lua/autorun/base_vehicles.lua

Should also explain that it's "Animation" or something

File: lua/entities/gmod_wire_expression2/core/custom/prop.lua [hunk 3]
```diff
@@ -216,12 +218,26 @@ e2function entity seatSpawn(string model, number frozen)
 	return PropCore.CreateProp(self,model,self.entity:GetPos()+self.entity:GetUp()*25,self.entity:GetAngles(),frozen,true)
 end
 
+e2function entity seatSpawn(string model, number frozen, string seat)
+	if not PropCore.ValidAction(self, nil, "spawn") then return NULL end
+	if model=="" then model = "models/nova/airboat_seat.mdl" end
+	if seat=="" then seat = "Seat_Airboat"
```
Comment (Vurv78) [inline lua/entities/gmod_wire_expression2/core/custom/prop.lua:224]: Missing an `end`

File: lua/entities/gmod_wire_expression2/core/custom/prop.lua [hunk 4]
```diff
@@ -216,12 +218,26 @@ e2function entity seatSpawn(string model, number frozen)
...
 e2function entity seatSpawn(string model, vector pos, angle rot, number frozen)
 	if not PropCore.ValidAction(self, nil, "spawn") then return NULL end
 	if model=="" then model = "models/nova/airboat_seat.mdl" end
 	return PropCore.CreateProp(self,model,Vector(pos[1],pos[2],pos[3]),Angle(rot[1],rot[2],rot[3]),frozen,true)
 end
 
+e2function entity seatSpawn(string model, vector pos, angle rot, number frozen, string seat)
+	if not PropCore.ValidAction(self, nil, "spawn") then return NULL end
+	if model=="" then model = "models/nova/airboat_seat.mdl" end
+	if seat=="" then seat = "Seat_Airboat"
```
Comment (Vurv78) [inline lua/entities/gmod_wire_expression2/core/custom/prop.lua:237]: Missing an `end`

**Review by Vurv78 — CHANGES_REQUESTED** [review]

**Comment by Derpius** [comment]
Consider testing your code or at least installing a linter 😉

**Comment by Hazeofdream** [comment]
> Consider testing your code or at least installing a linter 😉

tbh i was too assed to install a local version of wire and test my changes but from what I've seen here even wire devs do that too so its a human thing lmao

File: lua/entities/gmod_wire_expression2/core/custom/cl_prop.lua (hunk 2 above)
Comment (Hazeofdream) [reply lua/entities/gmod_wire_expression2/core/custom/cl_prop.lua:13]: i actually tried to search the github references for this in the actual wiki forthe "PlayerVehicleSpawn" of this and their were non, and oddly enough even this one gives different uses than what we do

**Pull request closed without merging**